django_lti_tool_provider/__init__.py

```diff
@@ -7,11 +7,14 @@
 logging.getLogger("").addHandler(logging.NullHandler())
 
 
+# pylint: disable=unused-argument
+# This class specifies method signatures; while pylint is intelligent enough to ignore unused args in abstractmethods
+# methods with default implementations are still reported. Hence we suppress unused-argument for entire class
 @add_metaclass(ABCMeta)
 class AbstractApplicationHookManager(object):
     """ Class that performs authentication and redirection for LTI views """
     @abstractmethod
-    def authentication_hook(self, request, user_id=None, username=None, email=None):
+    def authentication_hook(self, request, user_id=None, username=None, email=None, **kwargs):
```

Comment:
Let's assume that you have `role` parameter that is now in `**kwargs`, then in the future someone else adds additional non-optional parameter with the same name. And this parameter suddenly moves from `**kwargs` to `role` parameter which might introduce a bug that's not very easy to detect as code using `kwargs` would probably use `kwargs.get('role')` which will return `None` instead of exploding. 

I have had this issue couple of times, and I've seen couple of libraries regretting that decision.